### ISSUE TITLE
lhs of sampling can be complex expressions

### ISSUE THREAD

**Issue by jburroni** [deepppl/deepppl#4]
This is a valid program:
```
data {
 int N;
 vector[N] y;
}
parameters {
  real mu;
  real<lower=0.0> sigma;
}
mode {
  for (i in 1:10)
    (y[i] + mu)/sigma ~ normal(0, 1);
}
```

**Comment by gbdrt** [comment]
Do you know if stan does symbolic computation/rewrite? 
E.g.:
```
(y[i] + mu)/sigma ~ normal(0, 1);
-----------------------------------
y[i] ~ normal(mu, sigma)
```

**Comment by jburroni** [comment]
I don't know.
The things that worries me is that there are two different scenarios:

1. When there is data in the lhs
1. When there is no data in the lhs

The first one is not problematic at all. It's just a transformation of your data that we can apply.
The second scenario is more problematic as we may need to apply the inverse function of the expression to recover the sampled variables (if needed). I need to run more experiments to fully understand the implications of this.

**Comment by jburroni** [comment]
closed by #28
